data-prepper-plugins/mongodb/src/main/java/org/opensearch/dataprepper/plugins/mongo/stream/StreamWorker.java

```diff
@@ -208,20 +185,20 @@ public void processStream(final StreamPartition streamPartition) {
                             if (isCRUDOperation(operationType)) {
                                 final String record;
                                 if (OperationType.DELETE == operationType) {
-                                    record = document.getDocumentKey().toJson(writerSettings);
+                                    record = document.getDocumentKey().toJson(JSON_WRITER_SETTINGS);
                                 } else {
-                                    record = document.getFullDocument().toJson(writerSettings);
+                                    record = document.getFullDocument().toJson(JSON_WRITER_SETTINGS);
                                 }
                                 final long eventCreationTimeMillis = document.getClusterTime().getTime() * 1000L;
                                 final long bytes = record.getBytes().length;
                                 bytesReceivedSummary.record(bytes);
 
-                                checkPointToken = document.getResumeToken().toJson(writerSettings);
+                                checkPointToken = document.getResumeToken().toJson(JSON_WRITER_SETTINGS);
                                 // TODO fix eventVersionNumber
                                 final Event event = recordConverter.convert(record, eventCreationTimeMillis, eventCreationTimeMillis, document.getOperationTypeString());
-                                event.put(DEFAULT_ID_MAPPING_FIELD_NAME, event.get(MONGO_ID_FIELD_NAME, Object.class));
+                                // event.put(DEFAULT_ID_MAPPING_FIELD_NAME, event.get(DOCUMENTDB_ID_FIELD_NAME, Object.class));
```

Comment:
This will be configurable in follow up PR. If customer enables the configuration, we will write to customer provided field. Same for stream processing.